gnoi/gnoi.go

```diff
@@ -45,30 +44,27 @@ type Server struct {
 }
 
 // NewServer returns a new server that can be used by the mock target.
-func NewServer(privateKey crypto.PrivateKey, defaultCertificate *tls.Certificate, resetSettings *reset.Settings, notifyReset reset.Notifier, osSettings *os.Settings) (*Server, error) {
-	if defaultCertificate == nil {
```

Comment:
keep this logic within NewServer. Meaning, always create a private key and use it to generate a default certificate, then throw the private key away. This default certificate is only used to establish an encrypted session.
Pass the Settings untouched and directly to cert.NewManager.

Reply:
I kept the logic the same from before. I'm still generating the private key and passing the resulting certificate to the settings.